base_layer/core/src/transactions/transaction_components/unblinded_output.rs

```diff
@@ -242,7 +243,8 @@ impl UnblindedOutput {
                 .map_err(|_| TransactionError::RangeProofError(RangeProofError::ProofConstructionError))?
         };
 
-        let output = TransactionOutput::new_current_version(
+        let output = TransactionOutput::new(
+            self.version.clone(),
```

Comment:
I think this was a bug, as calling these functions would overwrite the given utxo version with whatever `get_current_version` would return